src/store/mutations.ts

```diff
@@ -0,0 +1,426 @@
+  state.stateModel.newRequestModel.nrData[key] = value
+}
+
+export const mutateNRDataByKey = (state: StateIF, kv: any) => {
+  Vue.set(
+    state.stateModel.newRequestModel.nr,
+    kv.key,
+    kv.value
+  )
+}
+
+export const mutateNrRequiredModalVisible = (state: StateIF, nrRequiredModalVisible: boolean) => {
+  state.stateModel.newRequestModel.nrRequiredModalVisible = nrRequiredModalVisible
+}
+
+export const mutatePickEntityModalVisible = (state: StateIF, pickEntityModalVisible: boolean) => {
+  state.stateModel.newRequestModel.pickEntityModalVisible = pickEntityModalVisible
+}
+
+export const mutatePickRequestTypeModalVisible = (state: StateIF, pickRequestTypeModalVisible: boolean) => {
+  state.stateModel.newRequestModel.pickRequestTypeModalVisible = pickRequestTypeModalVisible
+}
+
+export const mutatePriorityRequest = (state: StateIF, priorityRequest: boolean) => {
+  state.stateModel.newRequestModel.priorityRequest = priorityRequest
+}
+
+export const mutateRequestAction = (state: StateIF, action: string) => {
+  state.stateModel.newRequestModel.conversionType = ''
+  state.stateModel.newRequestModel.request_action_cd = action
+  if (action === 'MVE' && state.stateModel.newRequestModel.location === 'BC') {
+    state.stateModel.newRequestModel.location = 'CA'
+    state.stateModel.newRequestModel.entity_type_cd = 'XCR'
+  }
+}
+
+export const mutateRequestExaminationOrProvideConsent = (state: StateIF, { index, type, value }) => {
+  state.stateModel.newRequestModel.requestExaminationOrProvideConsent[index][type] = value
+}
+
+export const mutateShowActualInput = (state: StateIF, showActualInput: boolean) => {
+  state.stateModel.newRequestModel.showActualInput = showActualInput
+}
+
+export const mutateStats = (state: StateIF, stats: StatsI) => {
+  state.stateModel.newRequestModel.stats = stats
+}
+
+export const mutateSubmissionTabComponent = (state: StateIF, component: string) => {
+  enum Components {
+    EntityNotAutoAnalyzed,
+    NamesCapture,
+    ApplicantInfo1,
+    ApplicantInfo2
+  }
+  let tab = parseInt(Components[component])
+  state.stateModel.newRequestModel.submissionTabNumber = tab
+}
+
+export const mutateSubmissionTabNumber = (state: StateIF, submissionTabNumber: number) => {
+  state.stateModel.newRequestModel.submissionTabNumber = submissionTabNumber
+}
+
+export const mutateSubmissionType = (state: StateIF, submissionType: SubmissionTypeT) => {
+  state.stateModel.newRequestModel.submissionType = submissionType
+}
+
+export const mutateTabNumber = (state: StateIF, tabNumber: number) => {
+  state.stateModel.newRequestModel.tabNumber = tabNumber
+}
+
+export const mutateWaitingAddressSearch = (state: StateIF, appKV: WaitingAddressSearchI) => {
+  state.stateModel.newRequestModel.waitingAddressSearch = appKV
+}
+
+export const populateApplicantData = (state: StateIF) => {
+  for (let key in state.stateModel.newRequestModel.nr.applicants) {
+    Vue.set(
```

Comment:
FYI, this looks like a work-around to make some things reactive, which IMO is an architectural issue :( which we are hopefully in the process of fixing :)

>Due to limitations in JavaScript, there are types of changes that Vue cannot detect. However, there are ways to circumvent them to preserve reactivity.

https://vuejs.org/v2/guide/reactivity.html

Reply:
I was curious about the `Vue.set` approach. 
Wouldn't it be better to just build whatever object we wanted to set, then set it?

Reply:
Not sure exactly what the problem is here, nor how to fix it.

I do know there's a thing here that doesn't work as expected and `Vue.set` is a way to work around it. Maybe it's a setter problem. Maybe it's a module problem. Maybe it's a deep-watch problem. 🤷‍♂️